### PR TITLE
STYLE: Remove empty ResourceProbe legacy function GetSystemInformation()

### PR DESCRIPTION
**Pull request by N-Dekker**
Follow-up to:
"PERF: Remove SystemInformation data from ResourceProbe, fix issue #350"
Pull request https://github.com/InsightSoftwareConsortium/ITK/pull/351
Commit 747f3d143359cf15c3c4ed9e266125f96d198f8d, 2018-12-24

### CODE REVIEW

File: Modules/Core/Common/include/itkResourceProbe.hxx
Comment (dzenanz) [inline Modules/Core/Common/include/itkResourceProbe.hxx:563]: I think the policy is to remove deprecated things at the beginning of a new version. When starting 5.0 development, @hjmjohnson removed ITKv3 compatibility and everything else which was declared deprecated during 4.x. We declared this deprecated as of 5.0, so removing it in 5.2 already seems a bit fast.

Comment (N-Dekker) [reply Modules/Core/Common/include/itkResourceProbe.hxx:563]: > We declared this deprecated as of 5.0, so removing it in 5.2 already seems a bit fast.

Well, it was deprecated more than two years ago. In another decade. It feels like a long time... long before the corona pandemic...  😢 

Technically, I would prefer to prevent any user from calling `GetSystemInformation()` (as it does nothing useful), as well as from overriding the function (as it is not being called by internally by `ResourceProbe` anymore). The sooner the better. But if you think otherwise, I'll leave it up to you.

Comment (hjmjohnson) [reply Modules/Core/Common/include/itkResourceProbe.hxx:563]: @N-Dekker   I think we should start pushing for ITKv6.0 that only supports C++14 (or C++17), cmake 3.16+, python 3.6+ and make another round of deprecations.

:)

**Comment by hjmjohnson** [comment]
@N-Dekker Does building with ITK_LEGACY_REMOVE and ITK_FUTURE_LEGACY_REMOVE avoid the problem you are having?

**Comment by N-Dekker** [comment]
> @N-Dekker Does building with ITK_LEGACY_REMOVE and ITK_FUTURE_LEGACY_REMOVE avoid the problem you are having?

@hjmjohnson Thanks for asking Hans, I don't have a big problem with this empty (protected) `GetSystemInformation()` member function. I just proposed to remove it, because it isn't useful anymore. And it has been deprecated more than two years ago already.

But then Dženan (@dzenanz) explained that the function should still be there in ITKv5.2, so hopefully it can be removed before ITKv6! 

You may also just close this PR without merging, if you like.

**Review by hjmjohnson — APPROVED** [review]
We are past ITK 5.2 now.  I think this useless function can go.

**Comment by dzenanz** [comment]
I vote to wait until 5.3.0 final is tagged before merging this.

By the way, PerformanceBenchmarking is [currently broken on unix](https://open.cdash.org/viewBuildError.php?buildid=7456319):
`Modules/Core/Common/include/itkFixedArray.h:266:29: error: 'constexpr const ValueType& itk::FixedArray<TValue, VLength>::operator[](unsigned int) const' cannot be overloaded`
from `PerformanceBenchmarkingInformation.cxx:21`.

**Comment by N-Dekker** [comment]
> By the way, PerformanceBenchmarking is [currently broken on unix](https://open.cdash.org/viewBuildError.php?buildid=7456319):
> `Modules/Core/Common/include/itkFixedArray.h:266:29: error: 'constexpr const ValueType& itk::FixedArray<TValue, VLength>::operator[](unsigned int) const' cannot be overloaded`

Thanks @dzenanz But does it compile C++14? I see it also says:

> Modules/Core/Common/include/itkSize.h:481:27: error: 'MakeSize' function uses 'auto' type specifier without trailing return type

C++14 introduced that exactly! 😺

**Comment by dzenanz** [comment]
I created a [PR](https://github.com/InsightSoftwareConsortium/ITKPerformanceBenchmarking/pull/92) to check whether that is the issue.

**Comment by N-Dekker** [comment]
@dzenanz Now that ITK 5.3.0 is released, is it the right moment for this little PR?

**Comment by dzenanz** [comment]
I am fine with it. What do the others think?

**Comment by hjmjohnson** [comment]
@N-Dekker We are not at version 5.4.0.  Can you update this PR so it can be reviewed and finalized?

**Comment by hjmjohnson** [comment]
@N-Dekker I think it is time to resurrect and clear this PR.  Would you mind resolving the conflicts and poking the PR for re-review?

**Review by hjmjohnson — APPROVED** [review]

**Review by dzenanz — APPROVED** [review]

**Pull request merged**
